system/autoware_state_monitor/src/autoware_state_monitor_node/autoware_state_monitor_node.cpp

```diff
@@ -200,6 +200,45 @@ bool AutowareStateMonitorNode::onShutdownService(
   return true;
 }
 
+bool AutowareStateMonitorNode::onResetRouteService(
+  const std::shared_ptr<rmw_request_id_t> request_header,
+  const std::shared_ptr<std_srvs::srv::Trigger::Request> request,
+  const std::shared_ptr<std_srvs::srv::Trigger::Response> response)
+{
+  if (state_machine_->getCurrentState() != AutowareState::WaitingForEngage) {
+    response->success = false;
+    response->message = "Reset route can be accepted only under WaitingForEngage.";
+    return true;
+  }
+
+  state_input_.is_route_reset_required = true;
+
+  const auto t_start = this->now();
+  constexpr double timeout = 3.0;
+  while (rclcpp::ok()) {
+    rclcpp::spin_some(this->get_node_base_interface());
```

Comment:
In this line, The following error occurs.
```
what():  Node has already been added to an executor.
```

I think this line is not needed as this is not the main thread. (rclcpp::spin is already running)

Reply:
onShutdownService Function needs the same modification

Reply:
This was deeper issue.
The problem is, if we don't spin within the while loop, the other callback functions won't be called because the default executor is SingleThreadedExecutor, and the thread will be occupied by the while loop. 

Instead, I declared the executor as MultiThreadedExecutor to avoid the problem. I have assigned callback_groups so that two threads will be generated: one for the service callbacks and the other for subscriber/timer callbacks.